tests/test_fock.py

```diff
@@ -745,13 +753,21 @@ def circuit(a, phi):
 
         n = np.arange(cutoff)
 
+        # construct tape
+        circuit.construct([a, phi], {})
+
         # differentiate with respect to parameter a
-        res_F = circuit.jacobian([a, phi], wrt={0}, method="F").flat
+        circuit.qtape.trainable_params = {0}
+        res_F = circuit.qtape.jacobian(dev, method="numeric").flatten()
         expected_gradient = 2 * np.exp(-(a ** 2)) * a ** (2 * n - 1) * (n - a ** 2) / fac(n)
         assert np.allclose(res_F, expected_gradient, atol=tol, rtol=0)
 
+        # re-construct tape to reset trainable_params
+        circuit.construct([a, phi], {})
```

Comment:
I was a bit unsure whether this was the best way to do this, or whether this is actually something that should be fixed in PL core. The issue being that when setting e.g. `circuit.qtape.trainable_params = {0}` above and then calling `circuit.qtape.jacobian`, the second parameter (`1`) is given grad method `None`. Then, attempting to set `circuit.qtape.trainable_params = {1}` later doesn't work, since it only sets the grad method if the parameter has none from before. 

This happens in `JacobianTape._grad_method_validation`:

```python
if "grad_method" not in self._par_info[0]:
        self._update_gradient_info()
```

Reconstructing the tape was the easiest way of fixing this (other that perhaps removing the `if "grad_method" not in self._par_info[0]:` above).